### PR TITLE
Add thin wrapper on top of react-timeago's dateFormatter to support desired dates

### LINKED ISSUE #305: Timeago currently shows "2 years" if the project is 1 year and one day old
Always round down on years in timeago. For example on the project page.

### PR DESCRIPTION
**Pull request by piperchester**
## Description 
This fixes #305. I opted to go a step further, and include the number of days in the formatting. Happy to discuss the end result and iterate on whatever's desired! For example,

Screen Shot 2021-01-10 at 3 59 53 PM

Screen Shot 2021-01-10 at 3 59 39 PM

Please let me know if I missed any edge cases in the date calculation logic. I think #437 will help with more test data generation for local envs. 

## Also
- Fixed linting
- Updated test data script to use start dates beyond a year

### CODE REVIEW

File: README.md
Comment (piperchester) [inline README.md:89]: Added this snippet to start onboarding developers to more tests.

File: backend/climateconnect_api/management/commands/create_test_data.py
Comment (piperchester) [inline backend/climateconnect_api/management/commands/create_test_data.py:144]: @ddhanesha I added this as the default to create more variance in our test data. WDYT?

File: frontend/.eslintrc.js
Comment (piperchester) [inline frontend/.eslintrc.js:16]: @positiveimpact I bumped this to support optional chaining. I think we should probably update our lint config in the future, but seems low priority.

File: frontend/package.json
Comment (piperchester) [inline frontend/package.json:10]: My preference is to actually include this directly into lint, so that any time a contributor runs "lint" they automatically fix. I know that some people don't like that though so added this. Wdyt @positiveimpact?

File: frontend/pages/_app.js
Comment (piperchester) [inline frontend/pages/_app.js:47]: Lint fix.

File: frontend/src/components/browse/NoItemsFound.js
Comment (piperchester) [inline frontend/src/components/browse/NoItemsFound.js:13]: Remove unused for fixing lint.

File: frontend/src/utils/formatting.js
Comment (piperchester) [inline frontend/src/utils/formatting.js:26]: This is the core fix.

**Comment by piperchester** [comment]
@ reviewers: please review https://github.com/climateconnect/climateconnect/pull/438 first as it has some of these lint fixes.

File: backend/climateconnect_api/management/commands/create_test_data.py
Comment (positiveimpact) [reply backend/climateconnect_api/management/commands/create_test_data.py:144]: Sounds good! The more different we make our test data, the easier it will be to find missing edge cases while developing locally

File: frontend/package.json
Comment (positiveimpact) [reply frontend/package.json:10]: I have never tried this feature so I'm not sure how reliable it is. After all, we can still reverse a change it makes before committing. Still, I'd leave it as the `fix-lint` command for now and possibly move it to `lint` later.

File: frontend/src/utils/formatting.js
Comment (positiveimpact) [inline frontend/src/utils/formatting.js:19]: I'm a bit confused about the `epochMiliseconds - Date.now()` part - Shouldn't this result in `0`?
I just checked the react-timeago docs and they describe the fourth argument of the formatter like this:
`epochSeconds: The result of Date.now() or the result of a custom now prop.`

**Review by positiveimpact — APPROVED** [review]
LGTM overall. One very small general comment would be that I think it will now return `1 year and 0 days` rather than `1 year` if the time is exactly 1 year ago. 
Also, one line that I pointed out confused me a little bit but since it's taken directly from their react-timeago it must be correct.
Overall I like the idea of showing the date more exactly!
Approved! If the `1 year and 0 days` thing is correct please implement a change for it to say `exactly 1 year`or something like that. 
Afterwards you can merge without further review.

File: frontend/src/utils/formatting.js
Comment (piperchester) [reply frontend/src/utils/formatting.js:19]: Yeah good question! It's non-obvious (I believe the docs for `react-timeago` are incorrect as `Date.now()` returns milliseconds) and took a bit of double checking to work through. I opened a PR into the project to fix it: https://github.com/nmn/react-timeago/pull/151

I'll handle the 0 case, and add a couple more comments outlining it.

**Comment by piperchester** [comment]
Do you mind sanity checking this @positiveimpact locally? I think it's fine I'd love a final pair of eyes before landing it.

File: frontend/src/utils/formatting.js
Comment (positiveimpact) [inline frontend/src/utils/formatting.js:28]: `elapsedMilliseconds` isn't the diff between now and the date we're formatting. It's the total unix time elapsed until the date that we're formatting. I think you want the time elapsed between the date you're trying to format and now here.

**Review by positiveimpact — CHANGES_REQUESTED** [review]
There is one logic error in the new implementation, see my comment

File: frontend/pages/projects/[projectId].js
Comment (piperchester) [inline frontend/pages/projects/[projectId].js:12]: This was unused.

**Comment by piperchester** [comment]
Oof, this was a mind bender. The internal implementation of timeago was really confusing me with its inconsistent naming. 

Definitely was more work than I thought, but I think everything's working now:
Screen Shot 2021-02-03 at 6 17 29 PM

File: README.md
Comment (ddhanesha) [inline README.md:120]: Could rebase to master. We have made some changes to the README.md file

File: backend/climateconnect_api/management/commands/create_test_data.py
Comment (ddhanesha) [inline backend/climateconnect_api/management/commands/create_test_data.py:153]: Hmmmm... Makes sense. Should we do random start_date maybe? Don't think its necessary just a thought.

File: frontend/src/utils/formatting.js
Comment (ddhanesha) [inline frontend/src/utils/formatting.js:3]: nit: react-timeago

File: frontend/src/utils/formatting.js
Comment (ddhanesha) [inline frontend/src/utils/formatting.js:33]: Nit: I like comments, but some code is self explanatory. This could be concise comment.

**Review by positiveimpact — APPROVED** [review]
LGTM now!

File: backend/climateconnect_api/management/commands/create_test_data.py
Comment (piperchester) [reply backend/climateconnect_api/management/commands/create_test_data.py:153]: I like that idea for generating more entropy and randomness! We could revisit in a future PR.

File: frontend/src/utils/formatting.js
Comment (piperchester) [reply frontend/src/utils/formatting.js:33]: Great call out 🙏 agree this is overly verbose. I'll condense - thanks for highlighting!

File: frontend/src/utils/formatting.js
Comment (piperchester) [inline frontend/src/utils/formatting.js:3]:
```suggestion
 * react-timeago. The desired date is a format that's something like
```

Comment (piperchester) [reply frontend/src/utils/formatting.js:3]: Good catch @ddhanesha! Here's an example of the GitHub "suggestion" feature for spelling suggestions if you hadn't seen it before. It's super useful IMO.

**Comment by piperchester** [comment]
Thanks Dip and Chris - made some comment clarifications and updated `README.md`.

**Pull request merged**
